From cd874b9ddf968bea7ff4e6036f3a3898c752ba75 Mon Sep 17 00:00:00 2001
From: Yong Li <yong.b.li@intel.com>
Date: Fri, 26 Jun 2015 14:46:29 +0800
Subject: [PATCH] Mraa: Modify the I2C bus number for Minnowmax and DE3815
 platform; In IoT Linux 3.19. the default I2C bus is changed, as a quick
 solution, we modify the hard-coded I2C bus number in source code

[Mraa issue #198] -- https://github.com/intel-iot-devkit/mraa/issues/198
Upstream-Status: Pending

Signed-off-by: Yong Li <yong.b.li@intel.com>
---
 src/x86/intel_de3815.c     | 14 +++++++-------
 src/x86/intel_minnow_max.c |  2 +-
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/x86/intel_de3815.c b/src/x86/intel_de3815.c
index 2203a6c..76d4b3b 100644
--- a/src/x86/intel_de3815.c
+++ b/src/x86/intel_de3815.c
@@ -115,15 +115,15 @@ mraa_intel_de3815()
     b->pins[17].capabilites = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
 
     // BUS DEFINITIONS
-    b->i2c_bus_count = 2;
+    b->i2c_bus_count = 10;
     b->def_i2c_bus = 0;
-    b->i2c_bus[0].bus_id = 0;
-    b->i2c_bus[0].sda = 12;
-    b->i2c_bus[0].scl = 13;
+    b->i2c_bus[8].bus_id = 8;
+    b->i2c_bus[8].sda = 12;
+    b->i2c_bus[8].scl = 13;
 
-    b->i2c_bus[1].bus_id = 1;
-    b->i2c_bus[1].sda = 14;
-    b->i2c_bus[1].scl = 15;
+    b->i2c_bus[9].bus_id = 9;
+    b->i2c_bus[9].sda = 14;
+    b->i2c_bus[9].scl = 15;
 
     b->spi_bus_count = 1;
     b->def_spi_bus = 0;
diff --git a/src/x86/intel_minnow_max.c b/src/x86/intel_minnow_max.c
index 32eb098..48e8cac 100644
--- a/src/x86/intel_minnow_max.c
+++ b/src/x86/intel_minnow_max.c
@@ -32,7 +32,7 @@
 
 #define PLATFORM_NAME "MinnowBoard MAX"
 #define I2C_BUS_COUNT 10
-#define I2C_BUS_DEFAULT 7
+#define I2C_BUS_DEFAULT 8
 
 int arch_nr_gpios_adjust = 0x100;
 
-- 
2.1.4

